cmd/admin-trace.go

```diff
@@ -378,11 +458,161 @@ func matchingOpts(ctx *cli.Context) (opts matchOpts) {
+func rotateTraceLogsFile() {
+	defer func() {
+		// Close current file
+		if err := traceLogsFile.Close(); err != nil {
+			fatalIf(probe.NewError(err), "Unable to close current trace log file `"+traceLogsFilePath+"`")
+		}
+	}()
+	// Compress current file
+	if traceLogsCompress != "" {
+		// Remove raw file
+		defer func() {
+			if err := os.Remove(traceLogsFilePath); err != nil {
+				fatalIf(probe.NewError(err), "Unable to remove current trace log file `"+traceLogsFilePath+"`")
+			}
+		}()
+		switch traceLogsCompress {
+		case "gzip":
+			compressGzip()
+		case "zstd":
+			compressZstd()
+		default:
+			break
+		}
+	}
+}
+
+// gzip
+func compressGzip() {
+	compressFile, err := os.OpenFile(traceLogsFilePath+".gz", os.O_CREATE|os.O_RDWR, 0o644)
+	if err != nil {
+		compressFile.Close()
+		fatalIf(probe.NewError(err), "Unable to open trace log file archive for gzip `"+traceLogsFilePath+"`")
+	}
+	traceLogsFile.Seek(0, 0)
+	writer := gzip.NewWriter(compressFile)
+	_, err = io.Copy(writer, traceLogsFile)
+	if err != nil {
+		writer.Close()
+		compressFile.Close()
+		fatalIf(probe.NewError(err), "Unable to write trace log file archive for gzip `"+traceLogsFilePath+"`")
+	}
+	writer.Close()
+	compressFile.Close()
+}
+
+// zstandard
+func compressZstd() {
+	compressFile, err := os.OpenFile(traceLogsFilePath+".zst", os.O_CREATE|os.O_RDWR, 0o644)
+	if err != nil {
+		compressFile.Close()
+		fatalIf(probe.NewError(err), "Unable to open trace log file archive for zstd `"+traceLogsFilePath+"`")
+	}
+	traceLogsFile.Seek(0, 0)
+
+	if traceLogsFileZstdEncoder == nil {
+		traceLogsFileZstdEncoder, err = zstd.NewWriter(compressFile)
+		if err != nil {
+			traceLogsFileZstdEncoder.Close()
+			compressFile.Close()
+			fatalIf(probe.NewError(err), "Unable to create encoder for zstd `"+traceLogsFilePath+"`")
+		}
+	} else {
+		// Reuse encoder to reuse resources and avoid unnecessary allocation
+		traceLogsFileZstdEncoder.Reset(compressFile)
+	}
+
+	_, err = io.Copy(traceLogsFileZstdEncoder, traceLogsFile)
+	if err != nil {
+		traceLogsFileZstdEncoder.Close()
+		compressFile.Close()
+		fatalIf(probe.NewError(err), "Unable to copy encoder for zstd `"+traceLogsFilePath+"`")
+	}
+	traceLogsFileZstdEncoder.Close()
+	compressFile.Close()
+}
+
+// Create trace log file
+func createTraceLogsFile() *os.File {
+	// Create and open file if it does not exist
+	traceLogsFile, err := os.OpenFile(defineTraceLogsFile(), os.O_APPEND|os.O_CREATE|os.O_RDWR, 0o644)
+	if err != nil {
+		fatalIf(probe.NewError(err), "Unable to create trace log file `"+defineTraceLogsFile()+"`")
+	}
+	return traceLogsFile
+}
+
+// Derive new trace log file name
+func defineTraceLogsFile() string {
+	// Analyze pattern provided by user
+	pattern := regexp.MustCompile(`\[([^\[\]]+)\]`)
+	traceLogsDirectory := pattern.ReplaceAllStringFunc(traceLogsPattern, func(match string) string {
+		subMatches := pattern.FindStringSubmatch(match)
+		return traceLogstartTime.Format(subMatches[1])
+	})
```

Comment:
What does this do? I hate regexes.

Reply:
Ah, ok. Didn't catch that since there wasn't any docs:

`time.Now().Format("/tmp/minio/2006-01-02/minio-log-15-04-05.log")` will actually output what you expect, but with squares you can have numbers as part of the path. I think your approach is better, just doc it.